### PR TITLE
Change RequireHttpsAttribute to 302 by default

### LINKED ISSUE #4578: RequireHttps should be redirect 302 (not 301)
See https://github.com/aspnet/Mvc/issues/4561.

In this bug we just want to change the default, nothing more. In the other bug we'll make it configurable.

### PR DESCRIPTION
**Pull request by khellang**
Fixes #4578

### CODE REVIEW

**Comment by kichalla** [comment]
@Eilon

File: src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs [hunk 1]
```diff
@@ -76,15 +83,15 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
-                var newUrl = string.Concat(
-                    "https://",
-                    host.ToUriComponent(),
-                    request.PathBase.ToUriComponent(),
-                    request.Path.ToUriComponent(),
-                    request.QueryString.ToUriComponent());
+                var newUrl = UriHelper.Encode(
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs]: Why change this?

**Comment by Eilon** [comment]
@khellang thanks for the PR! For `release` we'd like to take only the change from 301 to 302. The rest we can do for `dev`. It's a bit late in the game for us to take any bigger change for `release` unfortunately.

File: src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs (hunk 1 above)
Comment (khellang) [inline src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs]: I thought it made sense to utilize the `UriHelper` from HttpAbstractions when I was through here anyway :smile:

**Comment by khellang** [comment]
@Eilon Alright. Updated to just change the default. Will send a PR with the property to dev when this is merged back :smile:

**Comment by Eilon** [comment]
Awesome, thanks for understanding!

**Comment by kichalla** [comment]
Thanks @khellang. This PR has now been merged into release and back to dev.

3c2bdfd368f1b0ed81e92f9287f43a58be06e6e0

**Pull request closed without merging**